### PR TITLE
fix: runner caching code that it uses

### PR DESCRIPTION
**Pull request by Skarlso**

### CODE REVIEW

**Comment by Skarlso** [comment]
```
ok      ocm.software/ocm/api/utils/template     12.614s
FAIL
make: *** [test] Error 1

```

It should have failed 😢

**Review by hilmarf — APPROVED** [review]

**Comment by Skarlso** [comment]
Alright, let's hope that it fails now...

File: .github/workflows/lint_and_test.yaml [hunk 1]
```diff
@@ -1,11 +1,7 @@
 name: lint-and-test
 
 on:
-  pull_request_target:
-  push:
-    branches:
-      - main
-  workflow_dispatch:
+  pull_request:
   workflow_call:
```
Comment (hilmarf) [inline .github/workflows/lint_and_test.yaml:5]: which one of these is actually required, when we want to run this during release?
https://github.com/open-component-model/ocm/blob/main/.github/workflows/release.yaml#L101

Comment (Skarlso) [reply .github/workflows/lint_and_test.yaml:5]: `workflow_call` I believe.

**Review by hilmarf — APPROVED** [review]

**Pull request merged**